lib/rules/no-useless-concat.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * @fileoverview disallow unncessary concatenation of template strings
+ * @author Henry Zhu
+ * @copyright 2015 Henry Zhu. All rights reserved.
+ * See LICENSE file in root directory for full license.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+/**
+ * Get's the right most node on the left side of a BinaryExpression with + operator.
+ * @param {ASTNode} node - A BinaryExpression node to check.
+ * @returns {ASTNode} node
+ */
+function getLeft(node) {
+    var left = node.left;
+    while (left.type === "BinaryExpression" && left.operator === "+") {
+        left = left.right;
+    }
+    return left;
+}
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+    return {
+        BinaryExpression: function(node) {
+            // account for the `foo + "a" + "b"` case
+            var left = getLeft(node);
+            var right = node.right;
+
+            if ((left.type === "Literal" || left.type === "TemplateLiteral") &&
+                (right.type === "Literal" || right.type === "TemplateLiteral") &&
+                node.operator === "+"
```

Comment:
Can you move this as a first check to short-circuit anything other then concatenation?